Sources/EasyPurchase/ProductInfoController.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  ProductInfoController.swift
+//
+//
+//  Created by BJIT on 19/9/23.
+//
+import Foundation
+import StoreKit
+
+// A protocol for handling product information retrieval.
+protocol ProductsInfoHandler {
+    // A method to fetch product information for a set of product IDs.
+    // - Parameters:
+    //   - productIds: A set of product IDs to fetch information for.
+    //   - completion: A closure that takes a `Products` object as its parameter
+    //     and is called when the product information retrieval is completed.
+    // - Returns: An `InAppProductRequest` object that can be used to start the request.
+    func fetchProductsInfo(_ productIds: Set<String>, completion: @escaping (Products) -> Void) -> InAppProductRequest
+}
+
+// A class that conforms to the `ProductsInfoHandler` protocol.
+class ProductInfoController: NSObject, ProductsInfoHandler {
+    // An instance of `InAppProductFetcherBuilder` used to build product retrieval requests.
+    var fetchProductBuilder = InAppProductFetcherBuilder()
```

Comment:
Make it private if not needed by others.

Reply:
Done. Thank you.